src/3d/terrain/qgsterraintexturegenerator_p.cpp

```diff
@@ -55,7 +55,8 @@ int QgsTerrainTextureGenerator::render( const QgsRectangle &extent, QgsChunkNode
   }
   mapSettings.setOutputSize( size );
 
-  QList<QgsMapLayer *> layers = mMap.layers();
+  // Map theme layer visibility taken into consideration in map settings
+  QList<QgsMapLayer *> layers = mapSettings.layers();
```

Comment:
Thank you for working on this. Indeed it does fixes the issue.

However, I think we could go a little further. If I understand the logic correctly (I don't know this part of the code): 
- `mapSettings` |layers` are built a first time in `baseMapSettings()`
- then, with your change, they are filtered to keep only the ones which don't have a renderer.

I think we should do this "no renderer" filtering in `baseMapSettings()` and remove all this `setLayers` logic from `render()`

Reply:
@ptitjano , sounds like a good idea, I shall adjust.

Reply:
@ptitjano , done.

Reply:
Looks good to me !